### DISCUSSION TITLE
How do I conditionally show a toast?

### DISCUSSION THREAD

**Discussion by agardnerIT** [reflex-dev/reflex#3858] [Q&A]
Brand new to reflex (looks fantastic btw!) so forgive me if this is obvious, but I'm missing something.

I want to conditionally show a toast if a boolean is `True`

```
rx.cond(FormState.f_submitted, rx.toast.success("Form submitted. Thanks!"))
```

First, [the docs state](https://reflex.dev/docs/library/dynamic-rendering/cond/): `The second component is optional and can be omitted. If it is omitted, nothing is rendered if the condition is False.` but with the above, I get:

```
ValueError: For conditional vars, the second argument must be set.
```

So I add a dummy `text` field to keep it happy:

```
rx.cond(FormState.f_submitted,
                rx.toast.success("foo"),
                rx.text(""))
```

and now get:

```
ValueError: Both arguments must be props.
```

## Reflex Version

```
reflex --version
0.5.10
```

**Comment by EzraJD** [comment]
Did you find a solution?  I am having the same issue.

**Comment by masenf** [comment]
probably what you would want to do here is `return` the toast from an event handler

```python
class MyState(rx.State):
    def handle_submit(self, form_data):
        # ... do form processing stuff
        return rx.toast.success("You submitted the form, congrats")

...

def index():
    return rx.form(
        ...,
        on_submit=MyState.handle_submit,
    )
```

In this case you display the toast imperatively when your event handler is done processing the form data.
